wordpress-mobile/WordPress-Android
# Add empty Site Intent Question screen in site creation flow

Closes [#16051](https://github.com/wordpress-mobile/WordPress-Android/issues/16051)

## Description

This PR adds a new (currently empty) step to the Site Creation flow, asking the users for their intent before presenting the site design options.

---

### Progress Status
The following checks from the issue acceptance criteria & other considerations are implemented:

- [x] The screen is displayed **only** when the `SITE_INTENT_QUESTION` development feature flag is enabled
  - [x] The screen is displayed as an new step in Site Creation flow
  - [x] The screen is displayed after tapping on 'CREATE A WORDPRESS.COM SITE' in the "Welcome to WordPress" screen.
- [x] Header (Back button, Skip Button, Title & Subtitle) is displayed
  - [x] Skip Button tap → Navigates forward to "Choose a Design" screen
  - [x] Back Button tap → Navigates back to:
    - [x] "Choose site" screen if logged in
    - [x] "Welcome to WordPress" screen if not
- [x] Unit Tests are added where applicable - see 1️⃣ below
- [ ] Events are tracked (Back button pressed, Skip button pressed, Screen viewed) - see 2️⃣ below
- [x] Back button on "Choose a Design" screens navigates back to the "What's your Website about" screen.

**1️⃣ I propose we tackle unit tests in a separate PR.**
There's little logic in the new code, except for firing events and making sure other parts pick them up as expected.

**2️⃣ I'm also proposing we merge this if we find it to be ready.**
We need this PR to progress with #16043 — that PR is responsible for adding the analytics events and it's [currently on hold](https://github.com/wordpress-mobile/WordPress-Android/pull/16043#issuecomment-1059021373) because we need at least the basic UI of the new screen to test it.


## To Test

<details>
<summary><h3>Test the existing flow</h3> (with the feature flag off)</summary>

1. Launch the app
2. From `My Site` screen tap the `⋁` icon next to the site title
3. On the `Choose a site` screen tap the `➕` menu button
4. Select `Create WordPress.com site`
5. **Verify** that you land on the `Choose a design` screen
</details>

### Test the new screen

1. Launch the app
2. Toggle the `SITE_INTENT_QUESTION` feature flag on

   <details>
   <summary>How to toggle the flag</summary>

   1. From `My Site` Go to `Me` → `App Settings` → `Debug settings`
   2. Scroll to `Features in development` section and tap on `SiteIntentQuestionFeatureConfig`
   3. Tap `RESTART THE APP` button
   </details>

3. From `My Site` screen tap the `⋁` icon next to the site title
4. On the `Choose a site` screen tap on `➕` menu button
5. Select `Create WordPress.com site`
6. **Verify** that you land on the `What's your website about?` screen

### Test interactions on the new screen

1. Tap on the `SKIP button`
2. **Verify** that you land on the `Choose a design` screen
3. Tap the `←` button or the device back button
4. **Verify** that you land on the `What's your website about?` screen
5. Tap again the `←` button or the device back button
6. **Verify** that you land on the `Choose a site` screen

<details>
<summary><h3>Test the new screen with a new account</h3></summary>

1. Launch the app
2. Log out ( From `My Site` Go to `Me` → `Log out of WordPress.com` → `LOG OUT` )
3. Tap 'Log in or sign up with WordPress.com`
4. Enter an e-mail address not registered with WordPress.com ([you can use the `+` sign with for Gmail](https://danq.me/2017/09/26/gmail-plus/))
5. Tap `Check Email` & pick an e-mail app
   1. In the e-mail app open the latest e-mail from WordPress.com
   2. Tap `Sign up to WordPress.com`
   3. Tap `Done` on first screen about your profile
   4. Tap `CREATE A WORDPRESS.COM SITE`
   5. **Verify** that you land on the `What's your website about?` screen
   6. Tap the `←` button or the device back button
   7. **Verify** that you land on the `My Site` screen
</details>

## Previews

| Portrait   | Landscape  |
| ---------- | ---------- |
| ![portrait](https://user-images.githubusercontent.com/4588074/158407714-32917ecb-b277-4504-aad2-7f7eb15e941e.jpg) | ![landscape](https://user-images.githubusercontent.com/4588074/158407746-30bb723b-116e-4c9c-a627-be97225e94d8.jpg) |

## Regression Notes

1. Potential unintended areas of impact
   - Existing flow with the feature flag off
   - Other steps in Site Creation when the feature is enabled

8. What I did to test those areas of impact (or what existing automated tests I relied on)
   - Manual Testing
   - Fixed title for the `Choose a domain` screen when the feature is enabled & added a test for it.

9. What automated tests I added (or what prevented me from doing so)
   - **`SiteCreationMainVM`**
      - `titleForDomainStepIsChooseADomain` - referenced above
      - `onSiteIntentSkippedPropagatedToWizardManager` - test the new method for handling the skip button
   - I thought about adding a test for the feature flag in ``
    

### PR submission checklist:
